pkg/ccl/backupccl/backup_test.go

```diff
@@ -1411,177 +1411,85 @@ into_db='restoredb', %s)`, encryptionOption), backupLoc1)
-	// produce. It merges contiguous ranges on the same node.
-	// It sorts ranges by start_key and counts the number of times the
-	// lease_holder changes by comparing against the previous row's lease_holder.
-	mergedRangeQuery := `
-WITH
-	ranges
-		AS (
-			SELECT
-				start_key,
-				lag(lease_holder) OVER (ORDER BY start_key)
-					AS prev_lease_holder,
-				lease_holder
-			FROM
-				[SHOW RANGES FROM TABLE data.bank WITH DETAILS]
-		)
-SELECT
-	count(*)
-FROM
-	ranges
-WHERE
-	lease_holder != prev_lease_holder
-	OR prev_lease_holder IS NULL;
-`
-
-	sqlDB.QueryRow(t, mergedRangeQuery).Scan(&totalExpectedBackupRequests)
-
-	backupTableID := sqlutils.QueryTableID(t, conn, "data", "public", "bank")
-
-	do := func(query string, check inProgressChecker) {
-		t.Logf("checking query %q", query)
-
-		var totalExpectedResponses int
-		if strings.Contains(query, "BACKUP") {
-			exportSpanCompleteCh = make(chan struct{})
-			// totalExpectedBackupRequests takes into account the merging that backup
-			// does of co-located ranges. It is the expected number of ExportRequests
-			// backup issues. DistSender will still split those requests to different
-			// ranges on the same node. Each range will write a file, so the number of
-			// SST files this backup will write is `backupRestoreDefaultRanges` .
-			totalExpectedResponses = totalExpectedBackupRequests
-		} else if strings.Contains(query, "RESTORE") {
-			// We expect restore to process each file in the backup individually.
-			// SST files are written per-range in the backup. So we expect the
-			// restore to process #(ranges) that made up the original table.
-			totalExpectedResponses = backupRestoreDefaultRanges
-		} else {
-			t.Fatal("expected query to be either a backup or restore")
-		}
-		jobDone := make(chan error)
-		allowResponse = make(chan struct{}, totalExpectedResponses)
-
-		go func() {
-			_, err := conn.Exec(query, localFoo)
-			jobDone <- err
-		}()
-
-		// Allow half the total expected responses to proceed.
-		for i := 0; i < totalExpectedResponses/2; i++ {
-			allowResponse <- struct{}{}
-		}
-
-		// Due to ExportRequest pagination, in the case of backup, we want to wait
-		// until an entire span has been exported before checking job progress.
-		if strings.Contains(query, "BACKUP") {
-			exportSpanCompleteCh <- struct{}{}
-			close(exportSpanCompleteCh)
+	defer cleanupFn()
+	var jobID jobspb.JobID
+	sqlDB.QueryRow(t, `BACKUP INTO 'userfile:///foo' WITH detached`).Scan(&jobID)
+	jobutils.WaitForJobToRun(t, sqlDB, jobID)
+
+	// Allow a few requests to go through.
+	blockCh <- struct{}{}
+	blockCh <- struct{}{}
+	blockCh <- struct{}{}
+
+	defer func() {
+		close(blockCh)
+		jobutils.WaitForJobToSucceed(t, sqlDB, jobID)
+		var finalProgress float32
+		sqlDB.QueryRow(t,
+			`SELECT fraction_completed FROM crdb_internal.jobs WHERE job_id = $1`,
+			jobID,
+		).Scan(&finalProgress)
+		require.Equal(t, float32(1), finalProgress)
+	}()
```

Comment:
I tried putting it in the testing hook but since we have concurrent backup processor workers I have to do some gymnastics to avoid a double close that added more code to this block. I've added a comment and left it as a defer for now.